src/client/Client.cc

```diff
@@ -12806,13 +12806,15 @@ int Client::_setxattr_check_data_pool(string& name, string& value, const OSDMap
     int64_t pool;
     try {
       pool = boost::lexical_cast<unsigned>(tmp);
-      if (!osdmap->have_pg_pool(pool))
-	return -CEPHFS_ENOENT;
-    } catch (boost::bad_lexical_cast const&) {
-      pool = osdmap->lookup_pg_pool_name(tmp);
-      if (pool < 0) {
-	return -CEPHFS_ENOENT;
+      if (osdmap->have_pg_pool(pool)) {
+          return 0;
       }
+    } catch (boost::bad_lexical_cast const&) {
+        ldout(cct, 15) << __func__ << ": pool in layout is not a pool id: " << tmp << dendl;
```

Comment:
You should move this debug out of the catch block. Because it will not be a pool id anyway.